app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/feature/AppTpRemoteFeatures.kt

```diff
@@ -71,11 +71,13 @@ class AppTpRemoteFeaturesStore @Inject constructor(
     private val stateAdapter: JsonAdapter<State> = moshi.newBuilder().add(KotlinJsonAdapterFactory()).build().adapter(State::class.java)
     private val listener = OnSharedPreferenceChangeListener { preferences, key ->
         coroutineScope.launch(dispatcherProvider.io()) {
-            val state = preferences.load(key)
-            if (state == null) {
-                togglesCache.remove(key)
-            } else {
-                togglesCache[key] = state
+            key?.let {
```

Comment:
I think we might need to handle the case when `key` is null. According to the docs, that signifies that the preferences were cleared and so perhaps the appropriate response is to clear the toggles cache. 

Not sure if there are gotchas to doing that here; @aitorvs thoughts?

Reply:
I think this would only matter if we called `clear` on the preferences so i wouldn't block on this, especially since we weren't handling that case before either from what i can see.